server/src/main/scala/org/ensime/server/WebServer.scala

```diff
@@ -2,84 +2,69 @@
 // License: http://www.gnu.org/licenses/gpl-3.0.en.html
 package org.ensime.server
 
-import java.io.File
+import scala.concurrent.{ Future, Promise }
 
-import akka.actor._
-import akka.http.scaladsl.marshallers.xml.ScalaXmlSupport
-import akka.http.scaladsl.model._
-import akka.http.scaladsl.server._
-import akka.stream._
-import akka.util.{ ByteString, Timeout }
-import com.google.common.io.Files
-import org.ensime.api._
-import org.ensime.jerky._
-import org.ensime.swanky._
+import io.netty.bootstrap.ServerBootstrap
+import io.netty.channel.nio.NioEventLoopGroup
+import io.netty.channel.socket.nio.NioServerSocketChannel
+import io.netty.channel.socket.SocketChannel
+import io.netty.channel.{ Channel, ChannelFuture, ChannelFutureListener, ChannelInitializer, EventLoopGroup, ChannelPipeline }
+import io.netty.handler.codec.http.websocketx.WebSocketServerProtocolHandler
+import io.netty.handler.codec.http.{ HttpObjectAggregator, HttpServerCodec }
+import io.netty.handler.logging.LoggingHandler;
 
-import scala.concurrent.Future
+import org.ensime.api.{ RpcRequestEnvelope, RpcResponseEnvelope }
+import org.ensime.core.DocJarReading
 
-trait WebServer {
-  implicit def system: ActorSystem
-  implicit def timeout: Timeout
-  implicit def mat: Materializer
+object WebServer {
```

Comment:
does this need to be a trait? Seems that having a singleton might not be good for testing.